R/print.data.table.R

```diff
@@ -30,13 +30,8 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
     # Other options investigated (could revisit): Cstack_info(), .Last.value gets set first before autoprint, history(), sys.status(),
     #   topenv(), inspecting next statement in caller, using clock() at C level to timeout suppression after some number of cycles
     SYS = sys.calls()
-    if (length(SYS) <= 2L ||  # "> DT" auto-print or "> print(DT)" explicit print (cannot distinguish from R 3.2.0 but that's ok)
-        ( length(SYS) >= 3L && is.symbol(thisSYS <- SYS[[length(SYS)-2L]][[1L]]) &&
-          as.character(thisSYS) == 'source') || # suppress printing from source(echo = TRUE) calls, #2369
```

Comment:
could you confirm you've tested this PR is a no-op for the #2369 case as well?

Reply:
Sorry, I shouldn't have removed that condition. Restored it now:

```
> .libPaths(c('data.table.Rcheck', .libPaths()))
writeLines('library(data.table)
DT = data.table(a = 1)
DT[ , a := 1]', 'test.R')
source("test.R", echo=TRUE)

> library(data.table)
data.table 1.16.99 IN DEVELOPMENT built 2024-12-03 19:02:16 UTC; aitap using 4 threads (see ?getDTthreads).  Latest news: r-datatable.com

> DT = data.table(a = 1)

> DT[ , a := 1]
>
```

Reply:
Great! Would it be easy to add this as a test to autoprint.R?

https://github.com/Rdatatable/data.table/blob/98cf24e635ad47ba02617929d2237fca263e2b0c/tests/autoprint.R